WCFDataService/Client/System/Data/Services/Client/Materialization/MaterializerEntry.cs

```diff
@@ -275,7 +275,7 @@ public static MaterializerEntry CreateEntryForLoadProperty(EntityDescriptor desc
         /// <returns>The materializer entry</returns>
         public static MaterializerEntry GetEntry(ODataEntry entry)
         {
-            return entry.GetAnnotation<MaterializerEntry>();
+            return entry == null ? null : entry.GetAnnotation<MaterializerEntry>();
```

Comment:
I checked all the caller of GetEntry, seems they call use the return entry directly. So it would throw NullReferenceException anyway. I think we can keep this change.